Cheng_gene_plots.R  plots for Extended Data Fig. 9a
Cheng_gene_plots_V1_newLayerDepths_correctOverlap.R  plots for Extended Data Fig. 9b,d
Depth_violin_V1_ITcells_subclass_2024-01-11.rmd  Violin plots of Extended Data Figure 7j; thanks to Rinaldo D'Souza, PhD, for creation of this script
Huntley2020_Exc_vs_Pv_genes_H3K27ac_ChIPcounts.R  Violin plots of fold difference in H3K27ac ChIP-seq signal in PV and total nuclei for PV and excitatory-enriched genes, Fig. 3c
Huntley2020_enriched_genes.R  Identification of PV-enriched and excitatory-enriched genes.
INTACT_cell_type_cCRE_TAD_meth_scatterplots.R  Scatter plots of cCRE methylation vs TAD methylation, Fig. 3e
INTACT_meth_MeCP2_WT_KO_pseudobulk_comparison.R  Scatter plot of gene-body methylation levels of WT and MeCP2 KO cells (Extended Data Fig. 2i) and INTACT vs snmC-seq methylation scatter plot (Extended Data Fig. 1d)
INTACT_meth_diamond_plots_rev.R  Diamond plots of genic methylation levels of INTACT subclasses and other plots, Fig. 1a,f,g and Extended Data Fig. 2g
INTACT_region_geneBodyTSSplus3kb_TSS_intra_extra_cCRE_meth_heatmap.R  Heatmaps of methylation level ratios for genomic elements associated with various genes, Fig. 2 and Extended Data Fig. 3
Li_nonPromoter_cCREs_1500bpWindows_Pv_H3K27ac_cellConfusionPaper_chip_plots.R  Script for H3K27ac fold change plots in Fig. 3 and Extended Data Fig. 4
PVGA_and_nonPVGA_cCRE_H3K27ac_ChIP_over_input_plots_wholeGenomeInputReads.R  Script for plots of H3K27ac ChIP signal in cCREs in Extended Data Fig. 4
PV_and_nonPV_cCRE_INTACT_PV_meth_boxplots.R  Boxplots of methylation of PV and non-PV cCREs in Fig. 3k and Extended Data Fig. 4j-k
Srgap1_Syt13_cCRE_meth_heatmap.R  Srgap1 and Syt13 heatmaps of Figure 3h
aggregate_methylation - folder containing scripts for producing precursors to aggregate methylation plots; thanks to Dennis Wu, PhD, for creation of these scripts
all_cs_effect_size_051822.R  DESeq2 on INTACT-isolated subclasses; thanks to Russell Moore, MD/PhD, for creation of this script
brain_tissue_global_meth_stripplots.R  Strip plots of global methylation levels of cerebellum, striatum, hypothalamus (Extended Data Fig. 2e) and gene expression fold change of high mCA genes in those tissues (Extended Data Fig. 2f)
cCRE_INTACT_methPerKB_boxplots.R  Boxplots of methylation level per kb, Fig. 3e and Extended Data Fig. 4b
cCRE_INTACT_meth_rep_stripplots.R  Strip plots of cCRE methylation in Fig. 3 and Extended Data Fig. 4
cCRE_MeCP2_ChIPcounts_WholeGenomeInputReads.R  Script for plots of MeCP2 ChIP signal in cCREs in Extended Data Fig. 4
cell_type_MeCP2_dysregulated_gene_lists_prefiltered.R  Identification of MeCP2-regulated genes in INTACT-isolated subclasses
cell_type_MeCP2dys_gene_cCRE_number_analyses.R  Boxplots of length and number of cCREs in MeCP2-repressed genes, Fig. 2f and Extended Data Fig. 3f
expression_of_Mecp2Reg_genes_regulated_by_cCREs.R
gene_body_INTACT_meth_boxplot_metaMR_resamp.R  Boxplot of methylation of core MeCP2-repressed genes, Extended Data Fig. 2d
gene_body_TSSplus3kb_and_flank_INTACT_meth_tables.R  Script for organizing gene-body and gene-flanking methylation level tables
gene_flank_meth_deciles.R  Diamond plots of number of MeCP2-repressed genes in deciles of methylation levels of INTACT-isolated subclasses, Fig. 2e
gene_flank_ridgeline_plots.R  Script for making ridgeline plots of methylation level in bins of various regions in INTACT-isolated subclasses
geo_submission_files_bisulfite.R  Helper script for organizing bisulfite GEO submission
geo_submission_files_checksum_check.R  Helper script for file checksum checking for GEO submission
global_meth_backgroundSub_INTACT_strip_plots.R  Strip plots of global methylation of INTACT-isolated subclasses, Extended Data Fig. 2h
homer_TF_diffExp_analysis.R  Boxplot of log2 fold change of gene expression of PV MeCP2-repressed genes, PV MeCP2-activated genes, transcription factors whose motifs are enriched in MeCP2-repressed PV cCREs, transcription factors whose motifs are enriched in MeCP2-repressed non-PV cCREs, and all other genes. Extended Data Fig. 4m
lambda_nonconversion_averaging.R  Calculating average bisulfite non-conversion rates using lambda phage
mecp2Het_merfish_composition_analysis.R  Composition analysis of WT and MeCP2 KO cells, Extended Data Fig. 7i
mecp2reg_genes_logFC_supplementary_tables.R  Helper script for organizing MeCP2-regulated genes, Table S2
merfish_cortex_hippocampus_analysis.R  Script for processing MERFISH data, including performing pseudobulkDGE between WT and MeCP2 KO cells of each cell type
merfish_cortex_hippocampus_qc_plots.R  Plots related to MERFISH, Fig. 5 and Extended Data Fig. 7, including UMAP, transcriptotyping, correlation matrix between experiments, INTACT RNA-seq vs MERFISH subclass gene expression scatter plots, number of cells in each subclass
merfish_cortex_hippocampus_subclass_nhood_DEG_plots.R  Violin plots and heatmap of type-DEGs in MERFISH experiments, Fig. 5h,i
merfish_expression_yao2021_marker_heatmap.R  Heatmap of marker genes in MERFISH data, Extended Data Fig. 7e
methylation_merging.py  Script for pooling methylation bed files
plot_enhancer_agg_meth_cellType_comparisons_100bpBins_2500bpFlanks_2lines.R  Aggregate methylation plots for cCREs in Extended Data Fig. 5c
plot_enhancer_agg_meth_cellType_comparisons_100bpBins_2500bpFlanks_nonConv.R  Aggregate methylation plots for cCREs in Fig. 2 and Extended Data Fig. 3
plot_metagene_agg_meth_cellType_comparisons_1kbBins_100kbRange_lessBorder_newBins_nonConv.R  Aggregate methylation plots for genes in Fig. 2 and Extended Data Fig. 3
plot_metagene_agg_meth_cellType_comparisons_1kbBins_100kbRange_lessBorder_only2.R  Aggregate methylation plots for genes in Extended Data Fig. 5c
pseudobulkDGE_by_layerDepth.R  Script for differential gene expression analysis through pseudobulkDGE for each layer depth, Fig. 6
subclass_gene_body_INTACT_meth_vs_INTACT_TPM_scatterplots.R  Scatter plots gene-body methylation vs gene expression, Extended Data Fig. 1e
subclass_gene_body_meth_corrPlots.R  Scatter plots of gene-body methylation of INTACT-isolated subclasses, Extended Data Fig. 1f 
summary_global_meth_brain_tissues_table.R  Summary script for organizing global methylation data of cerebellum, striatum, and hypothalamus
superficial_deep_L23_Cheng_gene_ratio_nonneg_zScore.R  Script for plots of gene expression of layer-specific genes across V1 of L2/3, Fig 6f
transcriptotype_summary_table_merfish.R  Summary table of number of cells in each transcriptotype
type_vs_type_pseudobulkDGE_analysis_INTACT.R  Boxplot comparing the WT fold differences in expression of type-specific DEGs between closely related types of the same subclass to fold differences between MeCP2 KO and WT cells of MeCP2-repressed genes or non-DEGs, Extended Data Fig. 7k
union_MeCP2dys_cCRE_MeCP2dys_gene_coding_prefilt5_nondedup_overlapAnalysis.R  Overlap heatmap of MeCP2-regulated cCREs and MeCP2-regulated genes
INTACT_long_mCAhigh_gene_TPM_boxplots.R  Violin plots of WT and MeCP2 KO gene expression of MeCP2-regulated genes, Extended Data Fig. 2a
INTACT_MA_plots.R  Log2 fold-difference in gene expression between WT INTACT-isolated subclasses, Extended Data Fig. 1c
